internal/k8s/advl4_controller.go

```diff
@@ -177,8 +177,22 @@ func (c *AviController) SetupAdvL4EventHandlers(numWorkers uint32) {
 
 	informer.GatewayInformer.Informer().AddEventHandler(gatewayEventHandler)
 	informer.GatewayClassInformer.Informer().AddEventHandler(gatewayClassEventHandler)
+}
 
-	return
+func (c *AviController) SetupNamespaceDeletionEventHandler(numWorkers uint32) {
+	nsHandler := cache.ResourceEventHandlerFuncs{
+		DeleteFunc: func(obj interface{}) {
+			ns, ok := obj.(*corev1.Namespace)
+			if !ok {
```

Comment:
we should try to fetch object whose status is `unknown` to handle stale entries. Similarly there is one more deleteFunc added at other place. We need to handle similar condition there,